domain/learningandworkprogress/src/main/kotlin/uk/gov/justice/digital/hmpps/domain/learningandworkprogress/induction/service/InductionScheduleService.kt

```diff
@@ -1,22 +1,47 @@
 package uk.gov.justice.digital.hmpps.domain.learningandworkprogress.induction.service
 
+import mu.KotlinLogging
 import uk.gov.justice.digital.hmpps.domain.learningandworkprogress.induction.InductionSchedule
+import uk.gov.justice.digital.hmpps.domain.learningandworkprogress.induction.InductionScheduleAlreadyExistsException
 import uk.gov.justice.digital.hmpps.domain.learningandworkprogress.induction.InductionScheduleHistory
 import uk.gov.justice.digital.hmpps.domain.learningandworkprogress.induction.InductionScheduleNotFoundException
 import uk.gov.justice.digital.hmpps.domain.learningandworkprogress.induction.InductionScheduleStatus
 import uk.gov.justice.digital.hmpps.domain.learningandworkprogress.induction.UpdatedInductionScheduleStatus
 import uk.gov.justice.digital.hmpps.domain.learningandworkprogress.induction.dto.UpdateInductionScheduleStatusDto
 import java.time.LocalDate
 
+private val log = KotlinLogging.logger {}
+
 class InductionScheduleService(
   private val inductionSchedulePersistenceAdapter: InductionSchedulePersistenceAdapter,
   private val inductionScheduleEventService: InductionScheduleEventService,
+  private val inductionScheduleDateCalculationService: InductionScheduleDateCalculationService,
 ) {
 
-  private val inductionScheduleDateCalculationService = InductionScheduleDateCalculationService()
-
   private val inductionScheduleStatusTransitionValidator = InductionScheduleStatusTransitionValidator()
 
+  /**
```

Comment:
This has been moved from `InductionService` and refactored to get the `CreateInductionScheduleDto` from the injected `InductionScheduleDateCalculationService`